config/confighttp/compression.go

```diff
@@ -51,18 +71,24 @@ var availableDecoders = map[string]func(body io.ReadCloser) (io.ReadCloser, erro
 		return gr, nil
 	},
 	"zstd": func(body io.ReadCloser) (io.ReadCloser, error) {
-		zr, err := zstd.NewReader(
-			body,
+		v := zstdReaderPool.Get()
+		var zr *zstd.Decoder
+		var err error
+		if v == nil {
+			// NOTE(tigrannajaryan):
 			// Concurrency 1 disables async decoding. We don't need async decoding, it is pointless
 			// for our use-case (a server accepting decoding http requests).
 			// Disabling async improves performance (I benchmarked it previously when working
 			// on https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23257).
-			zstd.WithDecoderConcurrency(1),
-		)
+			zr, err = zstd.NewReader(body, zstd.WithDecoderConcurrency(1))
```

Comment:
Is there a reason you haven't implemented the New() function in the pool to create the readers?